service/chain/protocol_state_test.go

```diff
@@ -0,0 +1,64 @@
+package chain_test
+
+import (
+	"math"
+	"os"
+	"testing"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/awfm9/flow-dps/service/chain"
+)
+
+var c *chain.ProtocolState
+
+func TestMain(m *testing.M) {
+	var err error
+	c, err = chain.FromProtocolState("./testdata")
```

Comment:
Not strictly related to this PR, but we've talked about it yesterday. `chain.FromProtocolState()` opens a badger database and obtains a lock on the directory, so multiple processes cannot open the same DB.

That's why `TestMain` was introduced, to setup the global state (open DB) on which individual tests rely on.

However, there is an option to open the DB as read-only:
https://pkg.go.dev/github.com/dgraph-io/badger/v2#Options.WithReadOnly

> When ReadOnly is true the DB will be opened on read-only mode. Multiple processes can open the same Badger DB. 

So it looks like it is possible, but would require a change in `FromProtocolState()`. It sounds like a more appropriate way for `FromProtocolState()` to function (rely on existing database, not create an empty one).